json_to_avro/avro_schema/mergeable_avro_schema.py

```diff
@@ -0,0 +1,573 @@
+import dataclasses
+import json
+from typing import (
+    List,
+    TypedDict,
+    Any,
+    Optional,
+    Literal,
+)
+from json_to_avro.json_to_avro.avroable_data import AvroableData, Json
+
+NullAvroType = Literal["null"]
+PrimitiveAvroType = Literal["string", "long", "double", "boolean"]
+
+
+class ArrayAvroType(TypedDict, total=False):
+    type: Literal["array"]
+    items: "AvroType"
+    default: Optional[Any]
+
+
+class AvroField(TypedDict, total=False):
+    name: str
+    type: "AvroType"
+    default: Optional[Any]
+
+
+class RecordAvroType(TypedDict, total=False):
+    type: Literal["record"]
+    name: str
+    fields: List[AvroField]
+    default: Optional[Any]
+
+
+class NullableAvroType(TypedDict, total=False):
+    type: list[Literal["null"] | "AvroType"]
+
+
+AvroType = NullAvroType | PrimitiveAvroType | ArrayAvroType | RecordAvroType
```

Comment:
All this typing needs massively revised or discarded. It doesn't currently cut it for mypy.